pkg/scheduler/framework/plugins/volumerestrictions/volume_restrictions.go

```diff
@@ -241,3 +323,66 @@ func New(_ runtime.Object, handle framework.Handle, fts feature.Features) (frame
 		enableReadWriteOncePod: fts.EnableReadWriteOncePod,
 	}, nil
 }
+
+// canBePreempt only verifies the volume, not the resource/affinity..
+func canBePreempt(nodeInfo *framework.NodeInfo, key string) bool {
+	namespace, pvcName := strings.Split(key, "/")[0], strings.Split(key, "/")[1]
+	anotherNodeInfo := nodeInfo.Clone()
+	for _, podInfo := range anotherNodeInfo.Pods {
+		pod := podInfo.Pod
+		if pod.Namespace != namespace {
+			continue
+		}
+		for _, vol := range pod.Spec.Volumes {
+			if vol.PersistentVolumeClaim.ClaimName == pvcName {
+				return true
+			}
+		}
+	}
+	return false
+}
+
+// preFilterState computed at PreFilter and used at Filter.
+// PreFilter traverses and compares whether the pvc referenced by each pod on each node
+// and the pvc of podToScheduled are the same pvc
+type preFilterState struct {
+	// All nodes that can match podToScheduled's all pvcs
+	// Used for fast-failure scheduling
+	nodesMatchPVC []*v1.Node
+	// If the AccessModes of the pvc used by a pod is ReadWriteOncePod,
```

Comment:
is it pods or PVCs? who do they belong to? the incoming Pod or all the Pods in the cluster?

Reply:
incoming Pod.The comment has been updated.